### PR TITLE
Create Tox action

### PR DESCRIPTION
**Pull request by t-persson**
### Applicable Issues

fixes: #34 

### Description of the Change

Add a Github action for running tox in the project on PR and on pushes

### Alternate Designs

I have decided to use python3.8 for the action and let Tox handle the python versions. Not sure if this is good or bad.

### Benefits

No more breaking tests. They are there for a reason!

### Possible Drawbacks

None

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Tobias Persson <tobias.persson@github.com>

### CODE REVIEW

**Comment by t-persson** [comment]
I have no idea if this works. Never tried creating Actions

File: main.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  push:
+    branches: [ master ]
```
Comment (magnusbaeck) [inline main.yml:9]: To make it easy to fine tune this file we should probably have it trigger on additional branches (or all?).

File: main.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      - uses: actions/checkout@v2
+
+      - name: Setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install Tox
+        run: pip install tox
+      - name: Run Tox
+        run: tox -e py
```
Comment (magnusbaeck) [inline main.yml:32]: Given
```
[tox]
envlist = py35
```
in tox.ini I wouldn't expect the line above to work. But why not just run `tox` and let tox.ini decide what to run?

**Review by magnusbaeck — APPROVED** [review]
I haven't used GitHub Actions so I have no idea whether it'll actually work, but it looks reasonable enough.

**Comment by t-persson** [comment]
Let's test it

**Pull request merged**